### PR TITLE
quantlib 1.29

### PR DESCRIPTION
**Pull request by skupr-anaconda**
Changelog: https://github.com/lballabio/QuantLib/releases
License: https://github.com/lballabio/QuantLib/blob/master/LICENSE.TXT
Requirements:
- https://www.quantlib.org/install.shtml
- https://www.quantlib.org/install/cmake.shtml
- https://github.com/lballabio/QuantLib/blob/QuantLib-v1.29/CMakeLists.txt

Actions:
1. Add `abs.yaml`: disable uploading to the target channel before the PR's approvals; the empty `build_parameters: - ""` will be needed if the artifacts will exist in the target channel, it will allow rewriting them by newer ones.
2. Reset `build/number` to `0`
3. Remove _autotools_ & `make` on `unix`
4. Add `ninja-base` on `unix`. It looks like the build process is faster with `ninja` locally and on Prefect 
5. Update `build.sh` & `bld.bat`
6. Disable test suite `DQL_BUILD_TEST_SUITE=OFF` and `ninja test` on unix because 1 test fails (failures are detected in the test module `Master Test Suite`) on `aarch64` and `s390x`
7. Add the compiler flag `-fno-fast-math`
8. Add pinnings for `boost-cpp` in `host` and `run`
9. Move the test command for `unix` to `build.sh` but comment on them there
10. Fix `home` url
11. Fix `description`

### CODE REVIEW

**Comment by skupr-anaconda** [comment]
An error on `s390x` and `aarch64`:
```
The following tests FAILED:
	  1 - quantlib_test_suite (Failed)
FAILED: CMakeFiles/test.util
```
It's not exactly clear why `quantlib_test_suite` fails

**Review by varlackc — COMMENTED** [review]
Just curious if we should remove the `abs.yaml` from the pull request.

**Comment by skupr-anaconda** [comment]
> Just curious if we should remove the `abs.yaml` from the pull request.

`aggregate_check: false` will be removed other keys will be enabled for uploading artifacts

**Review by akabanovs — APPROVED** [review]
LGTM

File: recipe/build.sh [hunk 1]
```diff
@@ -1,15 +1,40 @@
...
+    -DCMAKE_CXX_FLAGS="$CMAKE_CXX_FLAGS" \
+    -DQL_BUILD_EXAMPLES=OFF   \
+    -DQL_BUILD_TEST_SUITE=OFF \
+    -DQL_BUILD_BENCHMARK=OFF
+
+# Build.
+echo "Building..."
+ninja -j${CPU_COUNT} || exit 1
+
+# Perform tests.
```
Comment (danpetry) [inline recipe/build.sh]: This seems a little cryptic for someone in the future. Could be useful to say which of the below is logs with a bit more description and which should be uncommented.

**Review by danpetry — APPROVED** [review]
A minor nit but otherwise fine

File: recipe/build.sh (hunk 1 above)
Comment (skupr-anaconda) [reply recipe/build.sh]: I clarified a comment

**Pull request merged**